tests/devices/default_qubit/test_default_qubit_native_mcm.py

```diff
@@ -424,156 +425,141 @@ def circuit(x):
         _ = circuit([0.1, 0.2])
 
 
-# pylint: disable=not-an-iterable
-@pytest.mark.jax
+@pytest.mark.all_interfaces
+@pytest.mark.parametrize("interface", ["torch", "tensorflow", "jax", "autograd"])
 @pytest.mark.parametrize("mcm_method", ["one-shot", "tree-traversal"])
-@pytest.mark.parametrize("shots", [100, [100, 101], [100, 100, 101]])
-@pytest.mark.parametrize("postselect", [None, 0, 1])
-def test_sample_with_prng_key(mcm_method, shots, postselect):
-    """Test that setting a PRNGKey gives the expected behaviour. With separate calls
-    to DefaultQubit.execute, the same results are expected when using a PRNGKey"""
-    # pylint: disable=import-outside-toplevel
-    from jax.random import PRNGKey
-
-    dev = get_device(shots=shots, seed=PRNGKey(678))
-    params = [np.pi / 4, np.pi / 3]
-    obs = qml.PauliZ(0) @ qml.PauliZ(1)
+def test_finite_diff_in_transform_program(interface, mcm_method):
```

Comment:
I'm not entirely sure this is a mcm test if the diff method is explicitly specified as `"finite-diff"`.  What happens if `diff_method="best"`? Should we automatically switch to finite diff, or give a warning?

Reply:
Then we use parameter shift, but parameter shift is not correct if there is postselection or classical control.

Reply:
> Should we automatically switch to finite diff, or give a warning?

I think it makes sense to switch to finite diff.